vnext/Shared/InspectorPackagerConnection.cpp

```diff
@@ -0,0 +1,176 @@
+      return EventType::Connect;
+    if (0 == event.compare(InspectorMessage_eventName_disconnect))
+      return EventType::Disconnect;
+    assert(false && "Unknown event!");
+    std::abort();
+  }
+
+  static folly::dynamic getMessageForPackager(EventType eventType, folly::dynamic &&payload) {
+    folly::dynamic response = folly::dynamic::object;
+
+    switch (eventType) {
+      case EventType::GetPages:
+        response[InspectorProtocol::InspectorMessage_EVENT] = InspectorProtocol::InspectorMessage_eventName_getPages;
+        break;
+      case EventType::WrappedEvent:
+        response[InspectorProtocol::InspectorMessage_EVENT] =
+            InspectorProtocol::InspectorMessage_eventName_wrappedEvent;
+        break;
+      case EventType::Connect:
+        response[InspectorProtocol::InspectorMessage_EVENT] = InspectorProtocol::InspectorMessage_eventName_connect;
+        break;
+      case EventType::Disconnect:
+        response[InspectorProtocol::InspectorMessage_EVENT] = InspectorProtocol::InspectorMessage_eventName_disconnect;
+        break;
+      default:
+        assert(false && "Unknown event Type.");
+        std::abort();
+    }
+
+    response[InspectorProtocol::InspectorMessage_PAYLOAD] = std::move(payload);
+    return response;
+  }
+
+  static folly::dynamic getGetPagesResponseForPackagerPayload(
+      const std::vector<facebook::react::InspectorPage> &pages) {
+    folly::dynamic payload = folly::dynamic::array;
+    for (facebook::react::InspectorPage page : pages) {
+      folly::dynamic pageDyn = folly::dynamic::object;
+      pageDyn["id"] = page.id;
+      pageDyn["title"] = page.title;
+      pageDyn["vm"] = page.vm;
+
+      pageDyn["isLastBundleDownloadSuccess"] = true;
+      pageDyn["bundleUpdateTimestamp"] = "0";
+
+      payload.push_back(pageDyn);
+    }
+    return payload;
+  }
+
+  static folly::dynamic getVMResponseForPackagerPayload(int64_t pageId, std::string &&messageFromVM) {
+    folly::dynamic payload = folly::dynamic::object;
+    payload[InspectorProtocol::InspectorMessage_eventName_wrappedEvent] = messageFromVM;
+    payload[InspectorProtocol::InspectorMessage_PAGEID] = pageId;
+    return payload;
+  }
+
+  static folly::dynamic getVMResponseOnDisconnectForPackagerPayload(int64_t pageId) {
+    folly::dynamic payload = folly::dynamic::object;
+    payload[InspectorProtocol::InspectorMessage_PAGEID] = pageId;
+    return payload;
+  }
+};
+
+} // namespace
+
+RemoteConnection::RemoteConnection(int64_t pageId, const InspectorPackagerConnection &packagerConnection)
+    : m_packagerConnection(packagerConnection), m_pageId(pageId) {}
+
+void RemoteConnection::onMessage(std::string message) {
+  folly::dynamic response = InspectorProtocol::getMessageForPackager(
+      InspectorProtocol::EventType::WrappedEvent,
+      InspectorProtocol::getVMResponseForPackagerPayload(m_pageId, std::move(message)));
+  std::string responsestr = folly::toJson(response);
```

Comment:
Consider using `Windows::Data::Json` instead of folly::dynamic for JSON manipulation.

Reply:
My thoughts/preference here is just like the Websocket change that you suggested above.. I don't see anything wrong with folly::json for the usage here .. The lower level inspector implementation already uses folly_json anyways .. 
I understand that we should try to limit follly dependencies hoping someday we could get rid of it .. but i think it is not practical in near term .. 
I definitely want to rewrite using Windows::Data::Json, but i don't want to invest the time to rewrite the code and test it all over again. Hence, would like to defer it to another change after this PR gets in. It is alive for quite some time !
Hope you agree.